### PR TITLE
rec: Add some regression tests for RPZ over AXFR and IXFR

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by pieterlexis** [comment]
related: #4000

**Comment by rgacogne** [comment]
Would be nice to add to this (but doesn't prevent merging as it is):
- full `AXFR` after a few `IXFR` to make sure the rec handles that correctly
- check `RPZ` statistics via the `API` to make sure the serial and perhaps counters are OK

**Comment by rgacogne** [comment]
Full `AXFR` and `REST API` tests added.

File: regression-tests.recursor-dnssec/test_RPZ.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+                dns.rrset.from_text('zone.rpz.', 60, dns.rdataclass.IN, dns.rdatatype.SOA, 'ns.zone.rpz. hostmaster.zone.rpz. %d 3600 3600 3600 1' % newSerial),
+                dns.rrset.from_text('a.example.zone.rpz.', 60, dns.rdataclass.IN, dns.rdatatype.A, '192.0.2.1'),
+                dns.rrset.from_text('zone.rpz.', 60, dns.rdataclass.IN, dns.rdatatype.SOA, 'ns.zone.rpz. hostmaster.zone.rpz. %d 3600 3600 3600 1' % newSerial)
+                ]
+
+        elif message.question[0].rdtype == dns.rdatatype.IXFR:
+            oldSerial = message.authority[0][0].serial
+
+            if oldSerial != self._currentSerial:
+                print('Received an IXFR query with an unexpected serial %d, expected %d' % (oldSerila, self._currentSerial))
```
Comment (pieterlexis) [inline regression-tests.recursor-dnssec/test_RPZ.py]: s/Serila/Serial/

Comment (rgacogne) [reply regression-tests.recursor-dnssec/test_RPZ.py]: Good catch, thanks!

**Pull request merged**